### DISCUSSION TITLE
Flux Multi-cluster Multi-Repo

### DISCUSSION THREAD

**Discussion by tdiaque** [fluxcd/flux2#4889] [Q&A]
Hello

I understand that there are two different ways that FluxCD can be deployed based on this article https://control-plane.io/posts/fluxcd-architecture-overview/.

I also understand there's a strategy referred to as "Multi-tenancy" that I have looked into: https://github.com/fluxcd/flux2-multi-tenancy

I believe what I am looking for is similar to a combination of Hub & Spoke + Multi-Tenancy, the difference being is that each Spoke (Cluster) should be maintainable by a dev team from their own repository. Is this possible?

**Comment by stefanprodan** [comment]
The multi-tenancy lockdown works the same for local and spoke clusters. When reconciling a tenant's repo, Flux will use the SA assigned to that tenant in the spoke cluster.

**Comment by tdiaque** [comment]
Hi Stefan,

That's how I understand multi-tenancy should work. I am having issues though getting to the point where the spoke cluster has a GitRepository and deployments that will be pushed to it from the Hub cluster.

Here's my structure thus far (I believe my issue is with the tenants dir not having the GitRepository created in my Hub cluster)

```
- clusters/
       drt/
           apps/
               kustomization.yaml
               podinfo-values.yaml
            tenants/
                kustomization.yaml
- deploy/
       apps/
           kustomization.yaml
           podinfo.yaml
       tenants/
           kustomiation.yaml
           podinfo.yaml
 - hub/
        flux-system/
            gotk-components.yaml
            gottk-sync.yaml
            kustomization.yaml
        drt.yaml
 - tenants/
        drt/
            kustomization.yaml
            rbac.yaml
            sync.yaml
 ```           
 
 In the drt cluster repository. This is where as far as I understand, the podinfo-repository and release files would live to allow the drt cluster to have it's own repository and maintain the dependencies and apps that are deployed to this cluster.
```
 kustomization.yaml
 podinfo-release.yaml
 podinfo-repository.yaml
```
